### PR TITLE
docs: tweak postgresql compatibility blurbs

### PR DESCRIPTION
**Pull request by kay-kim**
Small tweaks to Postgres compatibility "strive for compatibility" blurb

### CODE REVIEW

File: doc/user/content/get-started/_index.md [hunk 1]
```diff
@@ -60,13 +60,21 @@ this exact purpose.
...
 
-Materialize follows the SQL standard (SQL-92) implementation, so you interact
-with it like any relational database: using SQL. You can build complex
-analytical workloads using **[any type of join](/sql/select/join/)** (including
-non-windowed joins and joins on arbitrary conditions), but you can also
-leverage exciting new SQL patterns enabled by streaming like
-[**Change Data Capture (CDC)**](/integrations/#databases),
-[**temporal filters**](/sql/patterns/temporal-filters/), and [**subscriptions**](/sql/subscribe/).
+You can interact with Materialize using SQL. You can build complex analytical
+workloads using **[various types of join](/sql/select/join/)** (including
```
Comment (morsapaes) [inline doc/user/content/get-started/_index.md:64]:
```suggestion
workloads using **[any type of join](/sql/select/join/)** (including
```

File: doc/user/content/get-started/_index.md [hunk 2]
```diff
@@ -60,13 +60,21 @@ this exact purpose.
...
+filters**](/sql/patterns/temporal-filters/), and
+[**subscriptions**](/sql/subscribe/).
+
+Materialize follows the SQL standard (SQL-92) implementation and strives for
+compatibility with the PostgreSQL dialect rather than aiming for compatibility
+with a specific version of PostgreSQL. This allows Materialize to add the latest
+PostgreSQL features even as some features of specific versions of PostgreSQL are
+missing. Performance for supported features may differ between Materialize and
+PostgreSQL, such as with [window
+functions](/transform-data/idiomatic-materialize-sql/appendix/window-function-to-materialize).
```
Comment (morsapaes) [inline doc/user/content/get-started/_index.md:77]:
```suggestion
Materialize follows the SQL standard (SQL-92) implementation and aims for
compatibility with the PostgreSQL dialect. It **does not** aim for
compatibility with a specific version of PostgreSQL. This means that
Materialize might support syntax from any released PostgreSQL version, but does
not provide full coverage of the PostgreSQL dialect. The implementation and
performance of specific features (like [window functions](/transform-data/idiomatic-materialize-sql/appendix/window-function-to-materialize))
might also differ, because Materialize uses an entirely different database engine
based on [Timely and Differential Dataflow](/get-started/#incremental-updates).
```

File: doc/user/content/get-started/_index.md [hunk 3]
```diff
@@ -60,13 +60,21 @@ this exact purpose.
...
 ### Standard SQL support
 
-Materialize follows the SQL standard (SQL-92) implementation, so you interact
-with it like any relational database: using SQL. You can build complex
-analytical workloads using **[any type of join](/sql/select/join/)** (including
-non-windowed joins and joins on arbitrary conditions), but you can also
-leverage exciting new SQL patterns enabled by streaming like
-[**Change Data Capture (CDC)**](/integrations/#databases),
-[**temporal filters**](/sql/patterns/temporal-filters/), and [**subscriptions**](/sql/subscribe/).
+You can interact with Materialize using SQL. You can build complex analytical
```
Comment (morsapaes) [inline doc/user/content/get-started/_index.md:63]:
```suggestion
The language used to interact with Materialize is **SQL**. You can build complex
analytical
```

Comment (morsapaes) [reply doc/user/content/get-started/_index.md:63]: "You can" makes it sound like you can use SQL, but you can also use something else.

File: doc/user/content/transform-data/_index.md [hunk 4]
```diff
@@ -9,18 +9,13 @@ menu:
     identifier: transform-overview
 ---
 
-With Materialize, you can use SQL to transform, deliver, and act on
```
Comment (morsapaes) [inline doc/user/content/transform-data/_index.md:12]: I wouldn't replace the blurb in this page. You could just link to the page above from "compatibility with the PostgreSQL dialect", rather than repeating the whole thing.

File: doc/user/content/transform-data/_index.md [hunk 5]
```diff
@@ -9,18 +9,13 @@ menu:
...
-join](/sql/select/join/) (including non-windowed joins and joins on arbitrary
-conditions). You can also leverage exciting new SQL patterns enabled by
-streaming, like [Change Data Capture
-(CDC)](/integrations/#databases), [temporal
-filters](/sql/patterns/temporal-filters/), and
-[subscriptions](/sql/subscribe/).
+Materialize follows the SQL standard (SQL-92) implementation and strives for
+compatibility with the PostgreSQL dialect rather than aiming for compatibility
+with a specific version of PostgreSQL. This allows Materialize to add the latest
+PostgreSQL features even as some features of specific versions of PostgreSQL are
```
Comment (morsapaes) [inline doc/user/content/transform-data/_index.md:15]: When talking about SQL, we should prefer to use the term "syntax" to "features" or "functionality". There's something to be said elsewhere about `pgwire`-compatibility that is unrelated (or, at least different) to SQL compatibility, too.

**Review by morsapaes — CHANGES_REQUESTED** [review]

File: doc/user/content/get-started/_index.md (hunk 3 above)
Comment (kay-kim) [reply doc/user/content/get-started/_index.md:63]: How about "You interact with Materialize using SQL"?  "The language used to interact ... is SQL" seems slightly awkward.

File: doc/user/content/get-started/_index.md (hunk 2 above)
Comment (kay-kim) [reply doc/user/content/get-started/_index.md:77]: Love the suggestion ❤️❤️ ❤️    Let me poke the people whose itch we're trying to scratch to check that they're good with this.

File: doc/user/content/transform-data/_index.md (hunk 5 above)
Comment (kay-kim) [reply doc/user/content/transform-data/_index.md:15]: There will be some one-off transgressions; e.g., will be adding a request a feature sentence ... and "request a syntax" seems clunky.

File: doc/user/layouts/partials/toc.html [hunk 6]
```diff
@@ -93,11 +93,7 @@ <h2 class="gradient_text dark">On this page</h2>
   // Scroll to our massaged offsets on a hashchange event, to override the
   // browser's automatic scroll to the unmassaged offset.
   $(window).on("hashchange", function () {
-    const id = window.location.hash.substring(1);
-    const offsets = headingOffsets();
-    const offset = offsets.find((o) => o.id == id);
-    if (offset) $(window).scrollTop(offset.offset - 10);
-    else $(window).trigger("scroll");
+   $(window).trigger("scroll");
```
Comment (kay-kim) [inline doc/user/layouts/partials/toc.html:96]: Had to tweak this -- otherwise, it wouldn't scroll correctly to the bottom.   The rewrite of the scroll logic is on my todo list - it's the highest of my non-content items.  Fingers crossed that maybe in December.

File: doc/user/layouts/shortcodes/materialize-postgres-compatibility.html [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+compatibility with the PostgreSQL dialect. It **does not** aim for compatibility
+with a specific version of PostgreSQL. This means that Materialize might support
+syntax from any released PostgreSQL version, but does not provide full coverage
+of the PostgreSQL dialect. To request a feature, see
+[Support](/support/#request-a-feature). The implementation and performance of
+specific features (like [window
+functions](/transform-data/idiomatic-materialize-sql/appendix/window-function-to-materialize))
+might also differ, because Materialize uses an entirely different database
+engine based on [Timely and Differential
+Dataflow](/get-started/#incremental-updates).
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/materialize-postgres-compatibility.html:11]:
```suggestion
Materialize follows the SQL standard (SQL-92) implementation and aims for
compatibility with the PostgreSQL dialect. It **does not** aim for
compatibility with a specific version of PostgreSQL. This means that
Materialize might support syntax from any released PostgreSQL version, but does
not provide full coverage of the PostgreSQL dialect. The implementation and
performance of specific features (like [window functions](/transform-data/idiomatic-materialize-sql/appendix/window-function-to-materialize))
might also differ, because Materialize uses an entirely different database
engine based on [Timely and Differential Dataflow](/get-started/#incremental-updates).

If you need specific syntax or features that are not currently supported in
Materialize, please submit a [feature request](/support/#request-a-feature).
```

File: doc/user/layouts/shortcodes/materialize-postgres-compatibility.html [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+compatibility with the PostgreSQL dialect. It **does not** aim for
+compatibility with a specific version of PostgreSQL. This means that
+Materialize might support syntax from any released PostgreSQL version, but does
+not provide full coverage of the PostgreSQL dialect. The implementation and
+performance of specific features (like [window functions](/transform-data/idiomatic-materialize-sql/appendix/window-function-to-materialize))
+might also differ, because Materialize uses an entirely different database
+engine based on [Timely and Differential Dataflow](/get-started/#incremental-updates).
+
+If you need specific syntax or features that are not currently supported in
+Materialize, please submit a [feature request](/support/#request-a-feature).
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/materialize-postgres-compatibility.html:11]:
```suggestion
Materialize, please submit a [feature request](/support/#requesting-a-feature).
```
